project/LanceWang12/readme.md

```diff
@@ -0,0 +1,102 @@
+# t-SNE implementation
+
+
+
+## Basic Information
+
+t-SNE (t-distributed stochastic neighbor embedding) is an dimension reduction algorithm.  It can be used in visualizing high dimension  data and feature reduction.  **By the way, I will compare my implementation with sklearn.**
+
+- github repository: https://github.com/LanceWang12/t-SNE-implementation
+
+
+
+## Problem to solve
+
+In the Big Data analysis, data is not only becoming bigger and bigger; but also becoming more and more complicated.  It causes a spectacular increase of the dimensionality of the data.  For example, the dimensionality of a set of images is the number of pixels in any image, which ranges from thousands to millions.
+
+Computer has no problem processing that many dimensions.  However, we are limited to three dimensions.  Computer still needs us, so we should find a way to effectively visualize high-dimensional data before handing it over to the computer.
+
+There are many methods to solve this problem.  For example, PCA, LDA, Kernel method, Random projection and so on.  t-SNE uses Gaussian distribution to approximate high dimentionality and uses t distribution to approximate low dimentionality.  It uses KL-divergency to compute similarity and uses gradient discent to get the optima.
+
+![](https://i.imgur.com/cBQO9oV.gif)
+
+(using MoviePy to produce animation)
+
+
+
+## System Architecture
+
+![](https://i.imgur.com/XOAw7qx.png)
+
+
+
+## API Description
+
+```c++
+// Initialize hyperparameter for t-SNE object
+int x[10][3] = [
+  [1, 2, 3],
+  [4, 5, 6],
+  ...
+]
+tSNE(
```

Comment:
It is not wrong to have many arguments although it is considered hard to maintain.  Having it as the first design gives you a starting point to refine.